future/src/ct/ct_image.cc

```diff
@@ -293,6 +293,33 @@ CtImageEmbFile::CtImageEmbFile(CtMainWin* pCtMainWin,
     update_label_widget();
 }
 
+const std::string CtImageEmbFile::_get_icon_for_file(const std::string& filename)
+{
+    //default icon
+    std::string result = "ct_file_icon";
+    char* content_type = g_content_type_guess(filename.c_str(), NULL, 0, NULL);
+    if (content_type != NULL)
```

Comment:
`std::unique_ptr` would be better here I think. Less chance of leaks if the code changes